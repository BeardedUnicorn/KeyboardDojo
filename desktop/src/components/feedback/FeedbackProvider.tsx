import { Snackbar, Alert, Slide } from '@mui/material';
import React, { createContext, useContext, useCallback, useState } from 'react';

import { TRANSITIONS } from '@/theme';

import type { AlertColor, SlideProps } from '@mui/material';
import type { FC, ReactNode } from 'react';

interface FeedbackContextType {
  showToast: (message: string, type?: AlertColor, duration?: number) => void;
  showSuccess: (message: string, duration?: number) => void;
  showError: (message: string, duration?: number) => void;
  showWarning: (message: string, duration?: number) => void;
  showInfo: (message: string, duration?: number) => void;
}

interface FeedbackState {
  open: boolean;
  message: string;
  type: AlertColor;
  duration: number;
}

const FeedbackContext = createContext<FeedbackContextType | undefined>(undefined);

function SlideTransition(props: SlideProps) {
  return <Slide {...props} direction="up" />;
}

export const FeedbackProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const [state, setState] = useState<FeedbackState>({
    open: false,
    message: '',
    type: 'info',
    duration: 4000,
  });

  const handleClose = useCallback(() => {
    setState((prev) => ({ ...prev, open: false }));
  }, []);

  const showToast = useCallback((
    message: string,
    type: AlertColor = 'info',
    duration: number = 4000,
  ) => {
    setState({
      open: true,
      message,
      type,
      duration,
    });
  }, []);

  const showSuccess = useCallback((message: string, duration?: number) => {
    showToast(message, 'success', duration);
  }, [showToast]);

  const showError = useCallback((message: string, duration?: number) => {
    showToast(message, 'error', duration);
  }, [showToast]);

  const showWarning = useCallback((message: string, duration?: number) => {
    showToast(message, 'warning', duration);
  }, [showToast]);

  const showInfo = useCallback((message: string, duration?: number) => {
    showToast(message, 'info', duration);
  }, [showToast]);

  const value = {
    showToast,
    showSuccess,
    showError,
    showWarning,
    showInfo,
  };

  return (
    <FeedbackContext.Provider value={value}>
      {children}
      <Snackbar
        open={state.open}
        autoHideDuration={state.duration}
        onClose={handleClose}
        TransitionComponent={SlideTransition}
        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
        sx={{
          '& .MuiSnackbar-root': {
            transition: TRANSITIONS.medium,
          },
        }}
      >
        <Alert
          onClose={handleClose}
          severity={state.type}
          variant="filled"
          elevation={6}
          sx={{
            minWidth: '280px',
            transition: TRANSITIONS.quick,
            '&:hover': {
              transform: 'translateY(-2px)',
            },
          }}
        >
          {state.message}
        </Alert>
      </Snackbar>
    </FeedbackContext.Provider>
  );
};

export const useFeedback = () => {
  const context = useContext(FeedbackContext);
  if (!context) {
    throw new Error('useFeedback must be used within a FeedbackProvider');
  }
  return context;
};
