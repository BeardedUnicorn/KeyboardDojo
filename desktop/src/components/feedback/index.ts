// Re-export FeedbackAnimation from gamification/notifications
export { FeedbackAnimation } from '../gamification/notifications';

// Re-export FeedbackProvider
export * from './FeedbackProvider';
