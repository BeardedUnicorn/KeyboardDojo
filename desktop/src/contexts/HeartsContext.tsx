import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';
import { heartsService } from '../services/heartsService';

// Interface for hearts context
interface HeartsContextType {
  currentHearts: number;
  maxHearts: number;
  nextRegenerationTime: number | null;
  timeUntilNextHeart: string;
  useHeart: () => boolean;
  addHearts: (amount: number) => void;
  refillHearts: () => void;
  isLoading: boolean;
}

// Create context with default values
const HeartsContext = createContext<HeartsContextType>({
  currentHearts: 0,
  maxHearts: 5,
  nextRegenerationTime: null,
  timeUntilNextHeart: '00:00',
  useHeart: () => false,
  addHearts: () => {},
  refillHearts: () => {},
  isLoading: true,
});

// Props for the provider component
interface HeartsProviderProps {
  children: ReactNode;
}

/**
 * Provider component for hearts/lives system
 */
export const HeartsProvider: React.FC<HeartsProviderProps> = ({ children }) => {
  const [heartsState, setHeartsState] = useState({
    currentHearts: 0,
    maxHearts: 5,
    nextRegenerationTime: null as number | null,
    lastUpdated: Date.now(),
  });
  const [timeUntilNextHeart, setTimeUntilNextHeart] = useState('00:00');
  const [isLoading, setIsLoading] = useState(true);

  // Initialize hearts service
  useEffect(() => {
    const initHearts = async () => {
      await heartsService.initialize();
      updateHeartsState();
      setIsLoading(false);
    };

    initHearts();

    // Set up timer to update the display
    const timer = setInterval(() => {
      updateHeartsState();
    }, 1000);

    return () => {
      clearInterval(timer);
      heartsService.cleanup();
    };
  }, []);

  // Update hearts state from service
  const updateHeartsState = () => {
    const state = heartsService.getState();
    setHeartsState(state);
    setTimeUntilNextHeart(heartsService.formatTimeUntilNextHeart());
  };

  // Use a heart
  const useHeart = (): boolean => {
    const result = heartsService.useHeart();
    updateHeartsState();
    return result;
  };

  // Add hearts
  const addHearts = (amount: number): void => {
    heartsService.addHearts(amount);
    updateHeartsState();
  };

  // Refill hearts
  const refillHearts = (): void => {
    heartsService.refillHearts();
    updateHeartsState();
  };

  // Context value
  const value: HeartsContextType = {
    currentHearts: heartsState.currentHearts,
    maxHearts: heartsState.maxHearts,
    nextRegenerationTime: heartsState.nextRegenerationTime,
    timeUntilNextHeart,
    useHeart,
    addHearts,
    refillHearts,
    isLoading,
  };

  return (
    <HeartsContext.Provider value={value}>
      {children}
    </HeartsContext.Provider>
  );
};

/**
 * Hook to use hearts context
 */
export const useHearts = (): HeartsContextType => {
  const context = useContext(HeartsContext);
  
  if (context === undefined) {
    throw new Error('useHearts must be used within a HeartsProvider');
  }
  
  return context;
};

export default HeartsContext; 