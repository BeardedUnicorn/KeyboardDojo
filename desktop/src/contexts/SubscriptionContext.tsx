import React, { createContext, useContext, useState, useEffect } from 'react';

import type { ReactNode , FC } from 'react';

interface SubscriptionContextType {
  hasPremium: boolean;
  isPremiumLoading: boolean;
}

const SubscriptionContext = createContext<SubscriptionContextType | undefined>(undefined);

export const SubscriptionProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const [hasPremium, setHasPremium] = useState(false);
  const [isPremiumLoading, setIsPremiumLoading] = useState(true);

  // Load subscription from localStorage on mount
  useEffect(() => {
    const loadSubscription = () => {
      try {
        setIsPremiumLoading(true);
        const savedSubscription = localStorage.getItem('user-subscription');

        if (savedSubscription) {
          const subscription = JSON.parse(savedSubscription);
          setHasPremium(subscription.hasPremium || false);
        }
      } catch (error) {
        console.error('Failed to load subscription:', error);
      } finally {
        setIsPremiumLoading(false);
      }
    };

    loadSubscription();

    // Simulate loading delay
    setTimeout(() => {
      setIsPremiumLoading(false);
    }, 1000);
  }, []);

  return (
    <SubscriptionContext.Provider
      value={{
        hasPremium,
        isPremiumLoading,
      }}
    >
      {children}
    </SubscriptionContext.Provider>
  );
};

export const useSubscription = (): SubscriptionContextType => {
  const context = useContext(SubscriptionContext);
  if (context === undefined) {
    throw new Error('useSubscription must be used within a SubscriptionProvider');
  }
  return context;
};
