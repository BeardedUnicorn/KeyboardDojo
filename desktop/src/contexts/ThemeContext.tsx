import React, { createContext, useContext, useState, useEffect } from 'react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import { createTheme, Theme } from '@mui/material/styles';
import { lightTheme, darkTheme } from '../theme';

type ThemeMode = 'light' | 'dark';

interface ThemeContextType {
  mode: ThemeMode;
  toggleTheme: () => void;
  theme: Theme;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export const useThemeContext = (): ThemeContextType => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useThemeContext must be used within a ThemeProvider');
  }
  return context;
};

interface ThemeProviderProps {
  children: React.ReactNode;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  // Get saved theme from localStorage or default to dark
  const getSavedTheme = (): ThemeMode => {
    const savedTheme = localStorage.getItem('themeMode');
    return (savedTheme as ThemeMode) || 'dark'; // Default to dark mode
  };

  const [mode, setMode] = useState<ThemeMode>(getSavedTheme());
  const [theme, setTheme] = useState<Theme>(mode === 'light' ? lightTheme : darkTheme);

  // Toggle between light and dark themes
  const toggleTheme = () => {
    const newMode = mode === 'light' ? 'dark' : 'light';
    setMode(newMode);
    localStorage.setItem('themeMode', newMode);
  };

  // Update theme when mode changes
  useEffect(() => {
    setTheme(mode === 'light' ? lightTheme : darkTheme);
  }, [mode]);

  const value = {
    mode,
    toggleTheme,
    theme,
  };

  return (
    <ThemeContext.Provider value={value}>
      <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
    </ThemeContext.Provider>
  );
}; 