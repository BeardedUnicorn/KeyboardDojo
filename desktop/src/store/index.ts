import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import { useDispatch, useSelector } from 'react-redux';

import { testReducer } from '@components/SentryReduxTest';
import {
  userProgressReducer,
  themeReducer,
  achievementsReducer,
  subscriptionReducer,
  gamificationReducer,
  curriculumReducer,
  settingsReducer,
  appReducer,
} from '@slices/index';
import { createSentryMiddleware } from '@utils/sentryRedux';

import { api } from './api';

import type { TypedUseSelectorHook } from 'react-redux';

// Create Sentry middleware
const sentryMiddleware = createSentryMiddleware();

// Create the store with middleware
export const store = configureStore({
  reducer: {
    // API reducer
    [api.reducerPath]: api.reducer,
    // Feature reducers
    userProgress: userProgressReducer,
    theme: themeReducer,
    achievements: achievementsReducer,
    subscription: subscriptionReducer,
    gamification: gamificationReducer,
    curriculum: curriculumReducer,
    settings: settingsReducer,
    app: appReducer,
    // Test reducer for Sentry integration testing
    test: testReducer,
  },
  // Adding middleware for development tools
  middleware: (getDefaultMiddleware) => 
    getDefaultMiddleware({
      serializableCheck: {
        // Ignore these action types
        ignoredActions: ['persist/PERSIST'],
      },
    }).concat(api.middleware, sentryMiddleware),
  // Enable Redux DevTools in development
  devTools: process.env.NODE_ENV !== 'production',
});

// Setup listeners for RTK Query
setupListeners(store.dispatch);

// Export types for dispatch and state
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// Create typed hooks
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

// Export the store as default
export default store; 
