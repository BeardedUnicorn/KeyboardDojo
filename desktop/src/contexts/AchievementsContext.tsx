import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';
import { achievementsService, AchievementProgress } from '../services/achievementsService';
import { userProgressService } from '../services/userProgressService';
import { AchievementNotification } from '../components';

export interface AchievementsContextType {
  achievements: AchievementProgress[];
  completedAchievements: AchievementProgress[];
  refreshAchievements: () => void;
  isLoading: boolean;
}

export const AchievementsContext = createContext<AchievementsContextType | null>(null);

export const useAchievements = () => {
  const context = useContext(AchievementsContext);
  if (!context) {
    throw new Error('useAchievements must be used within an AchievementsProvider');
  }
  return context;
};

interface AchievementsProviderProps {
  children: ReactNode;
}

export const AchievementsProvider: React.FC<AchievementsProviderProps> = ({ children }) => {
  const [achievements, setAchievements] = useState<AchievementProgress[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [unlockedAchievement, setUnlockedAchievement] = useState<AchievementProgress | null>(null);
  
  const refreshAchievements = () => {
    // Get current user progress
    const progress = userProgressService.getProgress();
    
    if (progress) {
      // Check for streak achievements
      const streakAchievements = achievementsService.checkAchievements(
        'streak', 
        progress.streakDays
      );
      
      // Check for practice achievements
      const practiceAchievements = achievementsService.checkAchievements(
        'practice', 
        progress.completedLessons.length
      );
      
      // Check for mastery achievements
      const masteryAchievements = achievementsService.checkAchievements(
        'mastery', 
        progress.completedModules.length
      );
      
      // Set the most recently unlocked achievement for notification
      const allUnlocked = [...streakAchievements, ...practiceAchievements, ...masteryAchievements];
      if (allUnlocked.length > 0) {
        setUnlockedAchievement(allUnlocked[allUnlocked.length - 1]);
      }
    }
    
    // Update achievements list
    setAchievements(achievementsService.getAchievements());
  };
  
  useEffect(() => {
    // Initialize achievements
    setIsLoading(true);
    setAchievements(achievementsService.getAchievements());
    setIsLoading(false);
    
    // Add listener for achievement updates
    const handleAchievementsUpdate = (updatedAchievements: AchievementProgress[]) => {
      setAchievements(updatedAchievements);
    };
    
    achievementsService.addListener(handleAchievementsUpdate);
    
    // Check for achievements on mount
    refreshAchievements();
    
    return () => {
      achievementsService.removeListener(handleAchievementsUpdate);
    };
  }, []);
  
  // Filter completed achievements
  const completedAchievements = achievements.filter(a => a.completed);
  
  return (
    <AchievementsContext.Provider
      value={{
        achievements,
        completedAchievements,
        refreshAchievements,
        isLoading
      }}
    >
      {children}
      
      {unlockedAchievement && (
        <AchievementNotification
          achievement={unlockedAchievement.achievement}
          onClose={() => setUnlockedAchievement(null)}
        />
      )}
    </AchievementsContext.Provider>
  );
}; 