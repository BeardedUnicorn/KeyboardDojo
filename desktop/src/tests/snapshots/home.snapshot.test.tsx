import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, expect, vi } from 'vitest';

import HomePage from '@pages/HomePage';

import { FeedbackProvider } from '../../components/feedback/FeedbackProvider';
import { SubscriptionProvider } from '../../contexts/SubscriptionContext';
import { store } from '../../store';

import type { ReactNode } from 'react';

// Mock Redux store
vi.mock('react-redux', () => ({
  ...vi.importActual('react-redux'),
  useSelector: vi.fn((selector) => {
    // Mock user progress state
    if (selector.name === 'selectUserProgress') {
      return {
        xp: 100,
        level: 2,
        streakDays: 5,
        hearts: { current: 5, max: 5 },
        completedLessons: [{ id: 'lesson-1', completedAt: new Date().toISOString() }],
      };
    }
    // Mock achievements state
    if (selector.name === 'selectAchievements') {
      return {
        achievements: [
          { id: 'achievement-1', title: 'First Lesson', unlockedAt: new Date().toISOString() },
        ],
        completedAchievements: [],
      };
    }
    // Mock theme state
    if (selector.name === 'selectTheme') {
      return 'light';
    }
    return undefined;
  }),
  useDispatch: () => vi.fn(),
}));

vi.mock('../../contexts/SubscriptionContext', () => ({
  SubscriptionProvider: ({ children }: { children: ReactNode }) => <div data-testid="subscription-provider">{children}</div>,
  useSubscription: () => ({
    hasPremium: false,
    isPremiumLoading: false,
  }),
}));

// Mock components
vi.mock('../../components/DailyGoal', () => ({
  default: () => <div data-testid="daily-goal">Daily Goal Component</div>,
}));

vi.mock('../../components/StreakDisplay', () => ({
  default: () => <div data-testid="streak-display">Streak Display Component</div>,
}));

vi.mock('../../components/LevelProgress', () => ({
  default: () => <div data-testid="level-progress">Level Progress Component</div>,
}));

vi.mock('../../components/RecentActivity', () => ({
  default: () => <div data-testid="recent-activity">Recent Activity Component</div>,
}));

vi.mock('../../components/QuickActions', () => ({
  default: () => <div data-testid="quick-actions">Quick Actions Component</div>,
}));

describe('HomePage Snapshot', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <FeedbackProvider>
            <SubscriptionProvider>
              <HomePage />
            </SubscriptionProvider>
          </FeedbackProvider>
        </Provider>
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
