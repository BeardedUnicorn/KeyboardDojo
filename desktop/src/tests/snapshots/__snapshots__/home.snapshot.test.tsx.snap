// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage Snapshot > renders correctly 1`] = `
<div>
  <div
    data-testid="theme-provider"
  >
    <div
      data-testid="user-progress-provider"
    >
      <div
        data-testid="achievements-provider"
      >
        <div
          data-testid="subscription-provider"
        >
          <div
            class="MuiBox-root css-19kzrtu"
          >
            <div
              class="MuiBox-root css-j4unr3"
            >
              <h1
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
              >
                Welcome to Keyboard Dojo
              </h1>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-1t4hig0-MuiTypography-root"
              >
                Improve your typing skills with personalized practice sessions
              </h6>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1qw2a80-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-uqzz15-MuiPaper-root-MuiCard-root"
                  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
                >
                  <div
                    class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
                  >
                    <div
                      class="MuiBox-root css-1coeexk"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1k3hahg-MuiSvgIcon-root"
                        data-testid="CodeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9.4 16.6 4.8 12l4.6-4.6L8 6l-6 6 6 6zm5.2 0 4.6-4.6-4.6-4.6L16 6l6 6-6 6z"
                        />
                      </svg>
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                      >
                        Shortcut Challenge
                      </h6>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
                    >
                      Master keyboard shortcuts for VS Code, IntelliJ, and Cursor with interactive challenges.
                    </p>
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth css-1hy71it-MuiButtonBase-root-MuiButton-root"
                      href="/shortcuts"
                      tabindex="0"
                    >
                      Start Challenge
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-uqzz15-MuiPaper-root-MuiCard-root"
                  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
                >
                  <div
                    class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
                  >
                    <div
                      class="MuiBox-root css-1coeexk"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1k3hahg-MuiSvgIcon-root"
                        data-testid="SchoolIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M5 13.18v4L12 21l7-3.82v-4L12 17zM12 3 1 9l11 6 9-4.91V17h2V9z"
                        />
                      </svg>
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                      >
                        Structured Curriculum
                      </h6>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
                    >
                      Follow a structured learning path with progressive lessons and challenges.
                    </p>
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1k9hkw1-MuiButtonBase-root-MuiButton-root"
                      href="/curriculum"
                      tabindex="0"
                    >
                      View Curriculum
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-uqzz15-MuiPaper-root-MuiCard-root"
                  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
                >
                  <div
                    class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
                  >
                    <div
                      class="MuiBox-root css-1coeexk"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1k3hahg-MuiSvgIcon-root"
                        data-testid="BarChartIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
                        />
                      </svg>
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                      >
                        Progress
                      </h6>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
                    >
                      View your typing statistics and track your improvement over time.
                    </p>
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1k9hkw1-MuiButtonBase-root-MuiButton-root"
                      href="/progress-dashboard"
                      tabindex="0"
                    >
                      View Progress
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-uqzz15-MuiPaper-root-MuiCard-root"
                  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
                >
                  <div
                    class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
                  >
                    <div
                      class="MuiBox-root css-1coeexk"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1k3hahg-MuiSvgIcon-root"
                        data-testid="EmojiEventsIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5h-2V3H7v2H5c-1.1 0-2 .9-2 2v1c0 2.55 1.92 4.63 4.39 4.94.63 1.5 1.98 2.63 3.61 2.96V19H7v2h10v-2h-4v-3.1c1.63-.33 2.98-1.46 3.61-2.96C19.08 12.63 21 10.55 21 8V7c0-1.1-.9-2-2-2M5 8V7h2v3.82C5.84 10.4 5 9.3 5 8m14 0c0 1.3-.84 2.4-2 2.82V7h2z"
                        />
                      </svg>
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                      >
                        Achievements
                      </h6>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-19b5ozw-MuiTypography-root"
                    >
                      Unlock achievements as you improve your typing skills and complete challenges.
                    </p>
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth css-1hy71it-MuiButtonBase-root-MuiButton-root"
                      href="/achievements"
                      tabindex="0"
                    >
                      View Achievements
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
